### DISCUSSION TITLE
How-To : hostile droid corpse

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#137] [General]
Looking for help with making a 'droid corpse' hostile.   

I've been using the tech_hub map exit from m4 as template.   I want to spawn a "less than dead' droid at start of Mission3
similar to the corpse at end of M4.

I spawned an android and set his flag for 'is_droid_corpse_on start' with a check.   
I set him right outside the tram on the ground since I could not get him to spawn in the transit office w/o elaborate proxies.

This gives me a downed inactive, but still "shootable" droid.  So his body can be shot and looted but he is otherwise 
non-functional. and his eyes are 'off'.  

There's a setting tied to a logic trigger that's too elaborate for this small exercise in the script but it's just 'finished spawning >droid>eyes_off'  and since I want them on I'm ont sure what that logic does.  There's a melee box entity with coordinates.  
I set up 3 around the coordinates of the downed droid or kinda randomly nearby. 
The melee box does not do anything so I suspect something is missed in setup.  

The M4 script is setup through premissionsetup and something else that is inapplicable again to this small exercise.  
The premissionsetup is alluding to inputs, but there are too many connections to decipher it.      I sorta of would like the 
droid tied to startup but if it complicates it I'd rather use a simple triggerbox.
I tried setting the droid alertness state to 2 too, but it does not seem to work on a droid that is "down".

Any ideas or insight will be useful.

**Comment by ghramsey** [comment]
I was able to move the corpse in the outro script of mission 4 (M07_PT2) either to the transit office or transit station, after painstakingly moving both the corpse and the meleecontext_droid_corpse boxes are the triggers for the grab sequence.  

At one point I was using a proxy and spawning the droid via the transitstation_section_1 zone, but I found this to not be necessary when used from the M07_PT2 script.   At that point I had a side effect; at the conclusion of the QTE the droid was duplicated.   So the conclusion is the initial corpse is only a visual prop. The melee boxes actually create an active droid which is transposed at the end of the QTE.   I removed the extra droid and this issue went away.
 
I seem to have proven I can move the corpse and manipulate his grab event.   But this is all in context of exit from M4 (M07_PT2 script.

I want to re-create this sequence in mission 3 (M06_PT2) script instead.  Hopefully w/o a dozen logic triggers that seem unnecessary?

**Reply by MattFiler** [reply]
Nice! Will be interested to see what success you have with re-creating it.

**Reply by ghramsey** [reply]
*In Hudson voice* "So far zippo!"  

 There are logic gates and triggers that seem to be irreproducable or transferrable in the M06_PT2 (M3 entry) script.
The droid won't even spawn using the triggers that work for the M07_PT2 (M4 exit) script.

I'm still missing something important. A prerequisite I suspect that ties everything.

**Comment by ghramsey** [comment]
Well.... I got it to work partially.    I created a triggerbox and duplicated the droids corpse and meleecontext box coordinate in systec
lobby in start of mission 3.  The droid trap worked perfectly.   Problem though I cannot move this corpse elsewhere and make it still
attack me.  Moved it to the transit station and the droid fails to spawn.  Tried a proxy, but I may be not setting the proxy up right or it 
may be the noted bug where proxies do not always work right.  No way to tell.
https://youtu.be/kZaIzdaQD60

**Comment by ghramsey** [comment]
So for the thing I must have been missing is the need to play through a new game.  I introduced a glitch or something else happened that made the tech support door that needs the keycard refuse to open.  It was displaying restore power.  
This error was extending to the transit office too.   

I started a new game from scratch and this issue was fixed proving it an error followed in the save file.  
As a side effect the droid corpse traps I setup now work.   

So my suspicion is that for some of these changes merely reloading a current save is insufficient to allow them to function.
That kinda sux because it's tedious to start a new game when play-testing.  However, from now on I'll need to remember that if something I edit fails to work then I need to make a new game playthrough.
